ilastik/applets/pixelClassification/suggestFeaturesDialog.py

```diff
@@ -767,157 +777,186 @@ def _auto_select_num_features(self, feature_order):
-                selected_feature_ids = selected_feature_ids[:n_selected]
-            else:
-                # make sure no more than n_features are requested
-                self.opGiniFeatureSelection.NumberOfSelectedFeatures.setValue(
-                    numpy.min([self._selection_params["num_of_feat"], self.n_features])
+        try:
+            all_features_active_matrix = numpy.zeros(user_defined_matrix.shape, "bool")
+            all_features_active_matrix[:, 1:] = True
+            all_features_active_matrix[0, 0] = True
+            all_features_active_matrix[1:, 0] = False  # do not use any other feature than gauss smooth on sigma=0.3
+            try:
+                self.opFeatureSelection.SelectionMatrix.setValue(all_features_active_matrix)
+            except FeatureSelectionConstraintError as err:
+                # Data dimensions too small along some of the axes, we fix it
+                msg = (
+                    "Some features cannot be used in automatic feature suggestion due to insufficient data dimensions.\n"
+                    "Automatic feature suggestion will continue, however:\n\n"
                 )
-                selected_feature_ids = self.opGiniFeatureSelection.SelectedFeatureIDs.value
-        elif self._selection_method == "filter":
-            if self._selection_params["num_of_feat"] == 0:
-                self.opFilterFeatureSelection.NumberOfSelectedFeatures.setValue(self.n_features)
-                selected_feature_ids = self.opFilterFeatureSelection.SelectedFeatureIDs.value
-
-                # now decide how many features you would like to use
-                # features are ordered
-                n_selected = self._auto_select_num_features(selected_feature_ids)
-                selected_feature_ids = selected_feature_ids[:n_selected]
+                # x-y: deselect scales completely where filters cannot be computed at all
+                if err.invalid_scales:
+                    all_features_active_matrix[:, numpy.isin(scales, err.invalid_scales)] = False
+                    msg += f"Features with sigmas {err.invalid_scales} will not be taken into account due to insufficient data dimensions in x-y.\n"
+                # z: calculate features in 2D for features that don't fit along z
+                if err.invalid_z_scales:
+                    compute_in_2d = numpy.copy(user_compute_in_2d)
+                    compute_in_2d[numpy.isin(scales, err.invalid_z_scales)] = True
+                    self.opFeatureSelection.ComputeIn2d.setValue(compute_in_2d.tolist())
+                    self.compute_in_2d_compat = compute_in_2d.tolist()
+                    msg += f"Calculating features at sigmas {err.invalid_z_scales} in 2D only."
+
+                logger.info(msg)
+
+                # suppress the popup in pytest:
+                if "PYTEST_CURRENT_TEST" not in os.environ:
+                    QMessageBox.information(self, "Not all features are compatible with your data!", msg)
```

Comment:
I am unhappy to have pytest-related shenanigans in production code, but alternative would be to keep a handle to the message box around in the dialog itself (I didn't manage to get a handle on it with `.activeModalWidget()`...